pandas/core/nanops.py

```diff
@@ -228,7 +228,9 @@ def _maybe_get_mask(
             # Boolean data cannot contain nulls, so signal via mask being None
             return None
 
-        if skipna:
+        if skipna or is_object_dtype(values.dtype):
+            # The masking in nanminmax does not work for object dtype, so always
```

Comment:
rather than do this, what exactly s the issue? 'does not work' is not very descriptive and generally we don't put comments like this, we simply fix it

Reply:
So what nanops.nanminmax appears to do when taking the min or max in the presence of missing values is to fill them with an appropriate infinite number that has the effect of ignoring those missing values (if we're taking the minimum replace with positive infinity, if we're taking the max replace with negative infinity). The problem is that this makes no sense for strings (there is as far as I know no "infinite string"), and that's why we get the error about comparing floats (infinity) and strings. The easiest workaround seems to be to mask them out instead.

To make things more complicated the `_get_values` function in `nanminmax` apparently doesn't bother to calculate a missing value mask when `skipna` is `False` because it's relying on the trick above working. Since it won't I'm making sure that we always get a mask for object dtypes.

Reply:
ok, then actually put this expl there. yes this might need a refactor for object type (which is out of scope for this PR) or at least backporting. so separate this out to another PR as its not trivially fixed here.

Reply:
Similarly as https://github.com/pandas-dev/pandas/pull/30982, I would personally rather have a separate implementation for one using masks instead of the filling logic of the nanops (with sharing helper functions where appropriate), instead of trying to fit it into the existing nanops code (which gives those special cases / complex if checks like the one below)

Reply:
@jorisvandenbossche Fair point for the string dtype, although I think some kind of logic like this would be necessary to fix `min / max` for object strings.

Reply:
> some kind of logic like this would be necessary to fix min / max for object strings.

Yep, indeed, that's what you mentioned as argument before as well for doing it in nanops, so the non-extension array object dtype would benefit as well. For this, we could also add a check for object dtype, and then calculate the mask and use the masked op implementation.

Reply:
Do you think it might also be worth trying to refactor nanminmax in nanops to use masking in general instead of the current filling approach (from what I could tell this was only really needed for arrays with more than one dimension)?

Reply:
yes